### PR TITLE
React disable button onclick

### LINKED ISSUE #118: Move all `this.props.history.push()` to ApplicationViews
Instead of have the individual components do the `history.push()` - which causes an unnecessary render - move them all to the functions that are defined in **`ApplicationViews`**.

Also change all `<a>` elements to buttons for actions in the app.

### PR DESCRIPTION
**Pull request by stevebrownlee**
Close #118 and #119

### CODE REVIEW

File: book-4-the-apprentice/chapters/REACT_DYNAMIC_ROUTING.md [hunk 1]
```diff
@@ -86,12 +201,71 @@ import { Link } from "react-router-dom";
...
 
-![animated image showing dynamic routing in action](./images/pZks6gwfvK.gif)
+## Refactor How Delete Works
+
+### Setup
+
+Add this import to the top of **`ApplicationViews`**.
+
+```js
+import { withRouter } from 'react-router'
```
Comment (JoeShep) [inline book-4-the-apprentice/chapters/REACT_DYNAMIC_ROUTING.md]: I"m not familiar with `withRouter`. What does that give us? I noticed in an earlier update you removed the `props` parameter from a Route compnent's callback. Is that related to this? Does `withRouter` allow us to skip the whole `{...props}` thing?

Comment (stevebrownlee) [reply book-4-the-apprentice/chapters/REACT_DYNAMIC_ROUTING.md]: `withRouter` which comes from React Router package allows a developer to use `history`, `location`, and `match` in a component that isn't being rendered as a child of a `<Route>`.

Since `ApplicationViews` is not a child of a `Route`, it needs to be used because the `deleteAnimal()` function was refactored to use `this.props.history.push()`. Since we are exporting `withRoute(ApplicationViews)`

Comment (stevebrownlee) [reply book-4-the-apprentice/chapters/REACT_DYNAMIC_ROUTING.md]: 1. I removed `props` from the lambda because it was confusing students.
1. All `withRouter` does is add the following namespaces to a component
    * `this.props.match`
    * `this.props.location`
    * `this.props.history`

This is done automatically if a component is returned from a `<Route>`, so you only use it for components that aren't.

```js
<Route path="/" render={() => <Joe/>} />
```

The `Joe` component will have `match`, `location`, and `history` added to `this.props` by the router package.
